src/2d/RenderSystem2D.cpp

```diff
@@ -39,6 +39,9 @@ namespace ari::en
 				// TODO: Set viewport
 				gfx::Viewport* pViewport = m_pFrameDataCurrent->Camera2dObj->GetViewport();
 
+				sx_rect ViewportRect = m_pFrameDataCurrent->Camera2dObj->GetViewportRect();
+				gfx::SetViewportRect(ViewportRect, false);
+
```

Comment:
What if the user didn't set the viewport rect at all.

Reply:
then we should set window rect as default, yeah?

Reply:
yes